src/fetch/transfer_file.py

```diff
@@ -137,6 +167,93 @@ def __str__(self) -> str:
 
 
 # pylint: disable=too-few-public-methods
+class SplitTransfers:
+    """
+    This class keeps the ERC20 and NATIVE token transfers Split.
+    Technically we should have two additional classes one for each token type.
+    """
+
+    def __init__(self, period: AccountingPeriod, mixed_transfers: list[Transfer]):
+        self.period = period
+        self.native, self.cow = [], []
+        for transfer in mixed_transfers:
+            if transfer.token_type == TokenType.NATIVE:
+                self.native.append(transfer)
+            elif transfer.token_type == TokenType.ERC20:
+                self.cow.append(transfer)
+            else:
+                raise ValueError(f"Invalid token type! {transfer.token_type}")
+        # Initialize empty overdraft
+        self.overdrafts: dict[Address, Overdraft] = {}
+        self.eth_transfers: list[Transfer] = []
+        self.cow_transfers: list[Transfer] = []
+
+    def _process_native_transfers(
+        self, indexed_slippage: dict[Address, SolverSlippage]
+    ) -> None:
+        # TODO - drain self.native into self.eth_transfers
+        for transfer in self.native:
+            solver = transfer.receiver
+            slippage: Optional[SolverSlippage] = indexed_slippage.get(solver)
+            if slippage is not None:
+                try:
+                    transfer.add_slippage(slippage)
+                except ValueError as err:
+                    name, address = slippage.solver_name, slippage.solver_address
+                    print(
+                        f"Slippage for {address}({name}) exceeds reimbursement: {err}\n"
+                        f"Excluding payout and appending excess to overdraft"
+                    )
+                    self.overdrafts[solver] = Overdraft(transfer, slippage, self.period)
+                    continue
+            self.eth_transfers.append(transfer)
+
+    def _process_token_transfers(self, cow_redirects: dict[Address, Vouch]) -> None:
+        # TODO - drain self.cow into self.cow_transfers
+        for transfer in self.cow:
+            solver = transfer.receiver
+            overdraft = self.overdrafts.get(solver)
+            if overdraft is not None:
+                cow_deduction = eth_in_token(
+                    TokenId.COW, overdraft.eth, self.period.end
+                )
+                print(f"Deducting {cow_deduction} from reward for {solver}")
+                transfer.amount -= cow_deduction
+                if transfer.amount < 0:
+                    print(
+                        "Overdraft exceeds COW reward! "
+                        "Excluding reward and updating overdraft"
+                    )
+                    overdraft.eth = token_in_eth(
+                        TokenId.COW, abs(transfer.amount), self.period.end
+                    )
```

Comment:
I think in an else statement, we should remove the overdraft. Otherwise, the overdraft will be printed later and it will be confusing for the user.
E.g. I was running the e2e test with 6000 cow, and I would get the following print:
```
Adjusting 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab(barn-0x) transfer by -0.32470 (slippage)
Slippage for 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab(barn-0x) exceeds reimbursement: Invalid adjustment TransferETH(receiver=0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab, amount=0.18536027477313313) by -0.32469736678953554
Excluding payout and appending excess to overdraft
Deducting 1694.5891470622905 from reward for 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab
Additional owed Overdraft(
    solver=0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab(barn-0x),
    period=2022-06-14-to-2022-06-21,
    owed=0.13933709201640243 ETH
)
```
Although the last overdraft was compensated with COW, it would still get printed as an overdraf

Reply:
Why wasn't this comment addressed?

Reply:
Will fix.

Reply:
Done: I will update the demonstrating that the overdrafts are properly removed. when they should be.